### PR TITLE
Add `JULIA_PKG_SERVER` to Environment Variables in docs

### PR DESCRIPTION
**Pull request by NHDaly**
I'm not really sure what to write in here. I'm not super familiar with this ENV variable, nor do I know how best to explain it.

I'd love some help with this!! Please feel free to edit this PR directly!!

I am opening this because we recently needed to disable the package server (we were having consistency issues, where the registry wasn't updating), and we couldn't find documented anywhere what the Environment Variable is to do that. We eventually found it in the julia slack. I'd like to add it here so others can find it. :) thanks!

### CODE REVIEW

**Review by Sacha0 — APPROVED** [review]
Superficially LGTM, but I know nothing about this, so requesting review from an authority :).

File: doc/src/manual/environment-variables.md [hunk 1]
```diff
@@ -156,6 +156,15 @@ The absolute path `REPL.find_hist_file()` of the REPL's history file. If
 $(DEPOT_PATH[1])/logs/repl_history.jl
 ```
 
+### `JULIA_PKG_SERVER`
+
+Used by `Pkg.jl`, for loading packages. By default, `Pkg` uses https://pkg.julialang.org to
+host julia packages. You can use this environment variable to disable the use of this
+package server, and instead access the packages hosted by GitHub, by setting:
```
Comment (giordano) [inline doc/src/manual/environment-variables.md:163]: I'm very far from being an authority in the Pkg land, but I think "loading" and "hosting" packages are used incorrectly here.  I'd say something along these lines
```suggestion
Used by `Pkg.jl`, for downloading packages and updating the registry. By default, `Pkg` uses `https://pkg.julialang.org` to
fetch Julia packages. You can use this environment variable to select a different server, or also to disable the use of the
PkgServer protocol, and instead access the packages directly from their hosts (GitHub, GitLab, etc.) by setting:
```

Comment (NHDaly) [reply doc/src/manual/environment-variables.md]: Cool, thanks for the change!

**Comment by fredrikekre** [comment]
`package_win32`:
```
ERROR: LoadError: failed to clone from https://github.com/JuliaRegistries/General.git, error: GitError(Code:ERROR, Class:Zlib, error inflating zlib stream)
Stacktrace:
  [1] pkgerror(msg::String)
    @ Pkg.Types C:\buildbot\worker\package_win32\build\usr\share\julia\stdlib\v1.7\Pkg\src\Types.jl:68
  [2] clone(io::IOContext{Base.PipeEndpoint}, url::String, source_path::String; header::String, credentials::Nothing, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Pkg.GitTools C:\buildbot\worker\package_win32\build\usr\share\julia\stdlib\v1.7\Pkg\src\GitTools.jl:119
  [3] (::Pkg.Registry.var"#28#29"{IOContext{Base.PipeEndpoint}, String, Pkg.Registry.RegistrySpec})(tmp::String)
    @ Pkg.Registry C:\buildbot\worker\package_win32\build\usr\share\julia\stdlib\v1.7\Pkg\src\Registry\Registry.jl:164
  [4] mktempdir(fn::Pkg.Registry.var"#28#29"{IOContext{Base.PipeEndpoint}, String, Pkg.Registry.RegistrySpec}, parent::String; prefix::String)
    @ Base.Filesystem .\file.jl:734
  [5] mktempdir
  ```
  and
  ```
  caused by: GitError(Code:ERROR, Class:Zlib, error inflating zlib stream)
Stacktrace:
  [1] macro expansion
    @ C:\buildbot\worker\package_win32\build\usr\share\julia\stdlib\v1.7\LibGit2\src\error.jl:110 [inlined]
  [2] clone(repo_url::SubString{String}, repo_path::String, clone_opts::LibGit2.CloneOptions)
    @ LibGit2 C:\buildbot\worker\package_win32\build\usr\share\julia\stdlib\v1.7\LibGit2\src\repository.jl:459
  [3] clone(repo_url::SubString{String}, repo_path::String; branch::String, isbare::Bool, remote_cb::Ptr{Nothing}, credentials::LibGit2.CachedCredentials, callbacks::Dict{Symbol, Tuple{Ptr{Nothing}, Any}})
    @ LibGit2 C:\buildbot\worker\package_win32\build\usr\share\julia\stdlib\v1.7\LibGit2\src\LibGit2.jl:583
  [4] clone(io::IOContext{Base.PipeEndpoint}, url::String, source_path::String; header::String, credentials::Nothing, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Pkg.GitTools C:\buildbot\worker\package_win32\build\usr\share\julia\stdlib\v1.7\Pkg\src\GitTools.jl:109
```

Could have been caused by bad network connection? restarting..

**Pull request merged**